Mathlib/Algebra/Group/Hom/Defs.lean

```diff
@@ -143,8 +143,9 @@ homomorphisms.
 
 You should also extend this typeclass when you extend `AddMonoidHom`.
 -/
-class AddMonoidHomClass (F M N : Type*) [AddZeroClass M] [AddZeroClass N] [FunLike F M N]
-  extends AddHomClass F M N, ZeroHomClass F M N : Prop
+class AddMonoidHomClass (F : Type*) (M N : outParam Type*)
```

Comment:
That we have to use `outParam`s in a subclass when `FunLike F M N` already has these `outParam`s is a bit weird, but it matches my experience (e.g. [here](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.60example.20.28p.20.3A.20P.29.20.3A.20Q.20.3A.3D.20p.60.20takes.200.2E25s.20to.20fail!/near/404670305)), and in particular it shouldn't cause any extra harm.

This change is extra good since it brings `AddMonoidHomClass` better in line with `MonoidHomClass` which already has `M N : outParam`.